src/main/java/seedu/address/logic/commands/interview/AcceptInterviewCommand.java

```diff
@@ -0,0 +1,80 @@
+package seedu.address.logic.commands.interview;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.List;
+
+import seedu.address.commons.core.DataType;
+import seedu.address.commons.core.Messages;
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.commands.Command;
+import seedu.address.logic.commands.CommandResult;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.applicant.Applicant;
+import seedu.address.model.interview.Interview;
+import seedu.address.model.position.Position;
+
+public class AcceptInterviewCommand extends Command {
+    public static final String COMMAND_WORD = "accept";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + " Parameters: INDEX (must be a positive integer)\n"
+            + "Example: " + COMMAND_WORD + " 1";
+
+    public static final String MESSAGE_ACCEPT_INTERVIEW_SUCCESS = "Accept Interview: %1$s";
+
+    private final Index targetIndex;
+
+    public AcceptInterviewCommand(Index targetIndex) {
+        this.targetIndex = targetIndex;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        List<Interview> lastShownList = model.getFilteredInterviewList();
+
+        if (targetIndex.getZeroBased() >= lastShownList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_INTERVIEW_DISPLAYED_INDEX);
+        }
+
+        Interview interviewToAccept = lastShownList.get(targetIndex.getZeroBased());
+
+        if (!model.isAcceptableInterview(interviewToAccept)) {
```

Comment:
Isn't isAcceptableInterview an Interview level method? Why is there a need to pass it through model?

Reply:
Thanks for your comment @YiHern-Lee. Passing it through the model will adhere to the current design of Logic -> Model -> ModelManager. Bypassing references and going straight to interviews seems to violate the purpose of the model manager, which is that everything should be controlled by the model interfaces.

Reply:
I actually disagree that this is the model's feature. It is Logic's responsibility to do the checks needed to perform some sort of alteration on the data. Checking for isAcceptableInterview seems to be the responsibility of Logic. Model should only handle changes that are pushed by Logic.